src/main/kotlin/io/bazel/worker/WorkerContext.kt

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package io.bazel.worker
+
+import io.bazel.worker.ContextLog.Granularity
+import io.bazel.worker.ContextLog.Granularity.INFO
+import io.bazel.worker.ContextLog.ScopeLogging
+import io.bazel.worker.Status.ERROR
+import java.io.ByteArrayOutputStream
+import java.io.Closeable
+import java.io.InterruptedIOException
+import java.io.PrintStream
+import java.nio.file.Path
+import java.util.logging.Level
+import java.util.logging.Logger
+import java.util.logging.SimpleFormatter
+import java.util.logging.StreamHandler
+
+/** WorkerContext encapsulates logging, filesystem, and profiling for a task invocation. */
+class WorkerContext private constructor(
+  private val name: String = Companion::class.java.canonicalName,
+  private val verbose: Granularity = INFO
+) : Closeable, ScopeLogging by ContextLogger(name, verbose.level, null) {
```

Comment:
Are you doing an anonymous delegate? That's... huh. I didn't realize you could do that. I'm used to delegating to a constructor parameter or property object, but not an anonymous instance like this.